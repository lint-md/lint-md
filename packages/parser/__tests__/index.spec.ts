import * as fs from 'fs';
import * as path from 'path';

const { parseMd } = require('../lib/lint-md-parser');

describe('test lint-md-parser', () => {
  const mdDemo = fs
    .readFileSync(path.resolve(__dirname, './common-demo.md'))
    .toString();

  test('test expose method (parseMd) is bundled as function', () => {
    expect(typeof parseMd).toStrictEqual('function');
  });

  test('test invoke parseMd', () => {
    expect(parseMd(mdDemo)).toMatchSnapshot();
  });
});
