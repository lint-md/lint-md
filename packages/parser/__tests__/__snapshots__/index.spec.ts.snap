// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test lint-md-parser test invoke parseMd 1`] = `
Object {
  "children": Array [
    Object {
      "position": Object {
        "end": Object {
          "column": 4,
          "line": 8,
          "offset": 68,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "yaml",
      "value": "categories:
  - Code
tags:
  - Java
  - IDEA
  - Spring Boot",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 14,
              "line": 10,
              "offset": 83,
            },
            "start": Object {
              "column": 3,
              "line": 10,
              "offset": 72,
            },
          },
          "type": "text",
          "value": "hello world",
        },
      ],
      "depth": 1,
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 10,
          "offset": 83,
        },
        "start": Object {
          "column": 1,
          "line": 10,
          "offset": 70,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 6,
              "line": 12,
              "offset": 90,
            },
            "start": Object {
              "column": 1,
              "line": 12,
              "offset": 85,
            },
          },
          "type": "text",
          "value": "你好 世界",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 6,
          "line": 12,
          "offset": 90,
        },
        "start": Object {
          "column": 1,
          "line": 12,
          "offset": 85,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Object {
                "end": Object {
                  "column": 16,
                  "line": 14,
                  "offset": 107,
                },
                "start": Object {
                  "column": 1,
                  "line": 14,
                  "offset": 92,
                },
              },
              "type": "text",
              "value": "www.example.com",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 16,
              "line": 14,
              "offset": 107,
            },
            "start": Object {
              "column": 1,
              "line": 14,
              "offset": 92,
            },
          },
          "title": null,
          "type": "link",
          "url": "http://www.example.com",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 18,
              "line": 14,
              "offset": 109,
            },
            "start": Object {
              "column": 16,
              "line": 14,
              "offset": 107,
            },
          },
          "type": "text",
          "value": ", ",
        },
        Object {
          "children": Array [
            Object {
              "position": Object {
                "end": Object {
                  "column": 37,
                  "line": 14,
                  "offset": 128,
                },
                "start": Object {
                  "column": 18,
                  "line": 14,
                  "offset": 109,
                },
              },
              "type": "text",
              "value": "https://example.com",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 37,
              "line": 14,
              "offset": 128,
            },
            "start": Object {
              "column": 18,
              "line": 14,
              "offset": 109,
            },
          },
          "title": null,
          "type": "link",
          "url": "https://example.com",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 43,
              "line": 14,
              "offset": 134,
            },
            "start": Object {
              "column": 37,
              "line": 14,
              "offset": 128,
            },
          },
          "type": "text",
          "value": ", and ",
        },
        Object {
          "children": Array [
            Object {
              "position": Object {
                "end": Object {
                  "column": 62,
                  "line": 14,
                  "offset": 153,
                },
                "start": Object {
                  "column": 43,
                  "line": 14,
                  "offset": 134,
                },
              },
              "type": "text",
              "value": "contact@example.com",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 62,
              "line": 14,
              "offset": 153,
            },
            "start": Object {
              "column": 43,
              "line": 14,
              "offset": 134,
            },
          },
          "title": null,
          "type": "link",
          "url": "mailto:contact@example.com",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 63,
              "line": 14,
              "offset": 154,
            },
            "start": Object {
              "column": 62,
              "line": 14,
              "offset": 153,
            },
          },
          "type": "text",
          "value": ".",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 63,
          "line": 14,
          "offset": 154,
        },
        "start": Object {
          "column": 1,
          "line": 14,
          "offset": 92,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Object {
                "end": Object {
                  "column": 16,
                  "line": 16,
                  "offset": 171,
                },
                "start": Object {
                  "column": 7,
                  "line": 16,
                  "offset": 162,
                },
              },
              "type": "text",
              "value": "Big note.",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 16,
              "line": 16,
              "offset": 171,
            },
            "start": Object {
              "column": 7,
              "line": 16,
              "offset": 162,
            },
          },
          "type": "paragraph",
        },
      ],
      "identifier": "1",
      "label": "1",
      "position": Object {
        "end": Object {
          "column": 16,
          "line": 16,
          "offset": 171,
        },
        "start": Object {
          "column": 1,
          "line": 16,
          "offset": 156,
        },
      },
      "type": "footnoteDefinition",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Object {
                "end": Object {
                  "column": 5,
                  "line": 18,
                  "offset": 177,
                },
                "start": Object {
                  "column": 2,
                  "line": 18,
                  "offset": 174,
                },
              },
              "type": "text",
              "value": "one",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 6,
              "line": 18,
              "offset": 178,
            },
            "start": Object {
              "column": 1,
              "line": 18,
              "offset": 173,
            },
          },
          "type": "delete",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 10,
              "line": 18,
              "offset": 182,
            },
            "start": Object {
              "column": 6,
              "line": 18,
              "offset": 178,
            },
          },
          "type": "text",
          "value": " or ",
        },
        Object {
          "children": Array [
            Object {
              "position": Object {
                "end": Object {
                  "column": 15,
                  "line": 18,
                  "offset": 187,
                },
                "start": Object {
                  "column": 12,
                  "line": 18,
                  "offset": 184,
                },
              },
              "type": "text",
              "value": "two",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 17,
              "line": 18,
              "offset": 189,
            },
            "start": Object {
              "column": 10,
              "line": 18,
              "offset": 182,
            },
          },
          "type": "delete",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 25,
              "line": 18,
              "offset": 197,
            },
            "start": Object {
              "column": 17,
              "line": 18,
              "offset": 189,
            },
          },
          "type": "text",
          "value": " tildes.",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 25,
          "line": 18,
          "offset": 197,
        },
        "start": Object {
          "column": 1,
          "line": 18,
          "offset": 173,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 25,
      "line": 18,
      "offset": 197,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
